ts/Series/Variwide/VariwideSeries.ts

```diff
@@ -0,0 +1,459 @@
+                right = xAxis.translate(
+                    (point.x as any) + (point.z as any),
+                    0 as any,
+                    0 as any,
+                    0 as any,
+                    1 as any
+                ) as any;
+            }
+
+            if (this.options.crisp) {
+                left = Math.round(left) - crisp;
+                right = Math.round(right) - crisp;
+            }
+
+            (point.shapeArgs as any).x = left;
+            (point.shapeArgs as any).width = Math.max(right - left, 1);
+
+            // Crosshair position (#8083)
+            point.plotX = (left + right) / 2;
+
+            // Adjust the tooltip position
+            if (!inverted) {
+                (point.tooltipPos as any)[0] =
+                    (point.shapeArgs as any).x +
+                    (point.shapeArgs as any).width / 2;
+            } else {
+                (point.tooltipPos as any)[1] =
+                    xAxis.len - (point.shapeArgs as any).x -
+                    (point.shapeArgs as any).width / 2;
+            }
+        }, this);
+
+        if (this.options.stacking) {
+            this.correctStackLabels();
+        }
+    }
+
+    // Function that corrects stack labels positions
+    public correctStackLabels(): void {
+        var series = this,
+            options = series.options,
+            yAxis = series.yAxis as StackingAxis,
+            pointStack,
+            pointWidth,
+            stack,
+            xValue;
+
+        series.points.forEach(function (
+            point: VariwidePoint
+        ): void {
+            xValue = point.x;
+            pointWidth = (point.shapeArgs as any).width;
+            stack = yAxis.stacking.stacks[(
+                series.negStacks &&
+                (point.y as any) < (
+                    options.startFromThreshold ?
+                        0 :
+                        (options.threshold as any)
+                ) ?
+                    '-' :
+                    ''
+            ) + series.stackKey];
+
+            if (stack) {
+                pointStack = stack[xValue as any];
+                if (pointStack && !point.isNull) {
+                    pointStack.setOffset(
+                        -(pointWidth / 2) || 0,
+                        pointWidth || 0,
+                        void 0,
+                        void 0,
+                        point.plotX
+                    );
+                }
+            }
+        });
+    }
+}
+
+/* *
+ *
+ * Prototype properties
+ *
+ * */
+interface VariwideSeries {
+    irregularWidths: boolean;
+    parallelArrays: Array<string>;
+    pointArrayMap: Array<string>;
+    pointClass: typeof VariwidePoint;
+    points: Array<VariwidePoint>;
+    relZ: Array<number>;
+    totalZ: number;
+    zData?: Array<number>;
+    correctStackLabels(): void;
+    postTranslate(
+        index: number,
+        x: number,
+        point?: VariwidePoint
+    ): number;
```

Comment:
With exception of pointClass all interface props not used in the extend call should go into the class pattern.